src/zjs_callbacks.c

```diff
@@ -149,6 +151,22 @@ static zjs_callback_id new_id(void)
     return id;
 }
 
+typedef struct deferred_work {
+    zjs_deferred_work callback;
+    u32_t length;  // length of user data
+    char data[0];  // user data
+} deferred_work_t;
+
+static void deferred_work_callback(void *handle, const void *args) {
+    const deferred_work_t *deferred = (const deferred_work_t *)args;
+
+    if (deferred->callback) {
```

Comment:
By this rationale I could potentially remove the NULL check we're looking at here. :) But that one can be caused by users of zjs_defer_work so I wanted to be a little more obvious. And it might get dropped in release build where DBG_PRINT does nothing.